### DISCUSSION TITLE
Ability to Request Changes to a Set/Achievement

### DISCUSSION THREAD

**Discussion by TheCommandoChicken** [RetroAchievements/RAWeb#2280] [Ideas]
I feel like there should be a better way to suggest changes to an achievement or set than just commenting and hoping the dev eventually listens. Having a more canonized way, like a button on a set's page that counts how many people think a revision is in order - like how requesting new sets works - would go a long way. Obviously just saying "this should be changed" wouldn't really help a dev know _how_ to change it, so there would probably also need to be a feedback field or something. Alternatively, being able to give feedback on individual achievements through a means similar to reporting actual issues would be good.

**Comment by televandalist** [comment]
It's probably still a ways away from happening since the RA forums still need a lot of work, but I know there are plans to be able to have multiple forum topics per game entry. I always pictured it being similar to Steam's "Discussions" section on a game page.

It would allow us to have:
- a pinned topic, which would likely be what is currently the "official forum topic" for an entry and would consist of important info regarding the set.
- topics for bugs/issues with the set, which I think would bring tickets down a bit since it would encourage actual dialogue between devs and players. I'm pretty vocal about how much I hate how the current ticket system works so this would be a huge plus in that regard 😆
- General help topics
- Suggestions
- etc etc

I think I speak for most of us devs who have been around for a while and have authored thousands of achievements when I say that feedback on individual achievements almost always gets buried due to the amount of notifications we tend to get. I regularly come across years-old achievement comments that I don't remember seeing before and feel like an ass for being so late to respond to them. I'd be in favor of killing achievement comments aside from the server logs if we were to have a more robust forum system.

Side note: having multiple pinned topics for a game could also minimize cases like the same achievement being reported to DevComp multiple times after it was already okay'd. 😓

(I'm just spitballing here)

**Reply by hb-robo** [reply]
I can definitely see how a long tail of development history can lead to notification overflow, that must be a pretty fundamental feedback problem for this kind of request.

However, from a user perspective I am pretty wary of even more infrastructure around forums and/or private messages as it doesn't feel "official" when interacting through them. Being entangled with social systems makes submissions like the OP describes very ignorable in my view because they inhabit the same on-page space and notification channels as casual conversation.

I think a more direct method would serve everyone a bit better, some kind of public submission board with firm, visible verdicts. Maybe they also could be voted on to show user interest, or each have their own comment box.

**Reply by foxsevent** [reply]
Just post a comment on the cheevo page, simple as that and as effective as any other method that you can think of.
No matter of the way it's gonna be done it's gonna be flooded with typical feedback "dis to hard should not be in mainset" at which point the dev will start ignoring it all even with some kind of voting system in place, at which you'd have to get DevComp or another dev involved to change the cheevo, delete it or also ignore it.

You can't make it "official" without making it someones responsibility or make them liable for it. 

> topics for bugs/issues with the set, which I think would bring tickets down a bit since it would encourage actual dialogue between devs and players. I'm pretty vocal about how much I hate how the current ticket system works so this would be a huge plus in that regard 😆

I think you forgot tele that most tickets are from people who can barely operate a PC that just goes "cheevo not unlock" where most of the time they did something wrong and now the dev has to play the guessing game of what's the issue when there probably never was one, and even if the dev is willing to talk to the user they would have to walk them like children the whole way thru which most people have no patience and time for, that is of course if either side is even ever gonna respond if the cheevo is actually broken and if ever the users provide a save state.

These are very valid issues that sadly there's no real solution IMHO for with RA being mostly charity work, maybe there could be solved if RA could hire people, maybe setup a Indian call customer service and maybe hire some devs/mods full time.

**Reply by wescopeland** [reply]
Given that we're zero-profit, this means **time is the most valuable resource of contributors.**

If contributors' tools such as the ticket manager can be enhanced/improved to enable those contributors to better leverage their time, quality should trend higher over a broad enough time horizon.

Many of these enhancements could be handled in the background transparently so they aren't relying on the end user to be technically sophisticated.

For example, tickets could automatically report the user's emulator, core, and/or ROM hash. We could have developer-only "debug RP" which dumped specific areas in memory that developers might be interested in learning about a user's session. There are all kinds of things we can do if we set out to take the time to build them.

A similar line of thinking could be taken about gathering feedback for achievements. We should be able to build tools that allow for feedback to be delivered to developers in a constructive, distilled, and concise way so developers don't have to spend time digging through user comments.
